homeassistant/components/tado/sensor.py

```diff
@@ -68,6 +84,114 @@ async def async_setup_entry(
         async_add_entities(entities, True)
 
 
+class TadoHomeSensor(TadoHomeEntity, Entity):
+    """Representation of a Tado Sensor."""
+
+    def __init__(self, tado, home_variable, is_metric):
+        """Initialize of the Tado Sensor."""
+        super().__init__(tado)
+        self._tado = tado
+
+        self.home_variable = home_variable
+
+        self._unique_id = f"{home_variable} {tado.home_id}"
+        self._is_metric = is_metric
+
+        self._state = None
+        self._state_attributes = None
+        self._tado_weather_data = self._tado.data["weather"]
+
+    async def async_added_to_hass(self):
+        """Register for sensor updates."""
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                SIGNAL_TADO_UPDATE_RECEIVED.format(
+                    self._tado.home_id, "weather", "data"
+                ),
+                self._async_update_callback,
+            )
+        )
+        self._async_update_home_data()
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._tado.home_name} {self.home_variable}"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._state_attributes
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self.home_variable == "temperature":
+            return TEMP_CELSIUS if self._is_metric else TEMP_FAHRENHEIT
+        if self.home_variable == "solar percentage":
+            return PERCENTAGE
+        if self.home_variable == "weather condition":
+            return None
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        if self.home_variable == "outdoor temperature":
+            return DEVICE_CLASS_TEMPERATURE
+        return None
+
+    @callback
+    def _async_update_callback(self):
+        """Update and write state."""
+        self._async_update_home_data()
+        self.async_write_ha_state()
+
+    @callback
+    def _async_update_home_data(self):
+        """Handle update callbacks."""
+        try:
+            self._tado_weather_data = self._tado.data["weather"]
+        except KeyError:
+            return
+
+        if self.home_variable == "outdoor temperature":
+            self._state = self.hass.config.units.temperature(
+                self._tado_weather_data["outsideTemperature"][
+                    "celsius" if self._is_metric else "fahrenheit"
+                ],
+                TEMP_CELSIUS if self._is_metric else TEMP_FAHRENHEIT,
```

Comment:
Just let core handle the conversion. Just return the current unit and value

Reply:
@elupus both units are provided, so I was just trying to avoid any conversions.
Anyway, I will change it in order to always return celsius.